src/tree.service.ts

```diff
@@ -24,6 +25,8 @@ export class TreeService {
   public nodeExpanded$: Subject<NodeExpandedEvent> = new Subject<NodeExpandedEvent>();
   public nodeCollapsed$: Subject<NodeCollapsedEvent> = new Subject<NodeCollapsedEvent>();
 
+  private controllers: Map<string | number, TreeController> = new Map();
```

Comment:
@lovery What will happen if a node gets renamed? Will controller for the old key still live in this map?
Also in case of destroy (`OnDestroy` interface) we should remove itself from this map in order to prevent potential memory leak